packages/solana/lib/src/encoder/signature.dart

```diff
@@ -1,11 +1,14 @@
 import 'package:cryptography/cryptography.dart' as crypto;
+import 'package:solana/src/base58/encode.dart';
 import 'package:solana/src/common/byte_array.dart';
 
 class Signature extends ByteArray {
   Signature.from(crypto.Signature signature) : _data = signature.bytes;
 
   final ByteArray _data;
 
+  String encodeBase58() => base58encode(_data.toList(growable: false));
```

Comment:
Lets call this `toBase58()` as do most of the libraries.

[web3](https://solana-labs.github.io/solana-web3.js/classes/PublicKey.html#toBase58) [Go](https://github.com/portto/solana-go-sdk#new-account--get-some-airdrop) [Java](https://github.com/p2p-org/solanaj/blob/f83f95b9b484b06495456d592659a1c90eb0448a/src/main/java/org/p2p/solanaj/core/PublicKey.java#L45).

Reply:
I do know that this is not the `PublicKey` class but it is basically the same idea. We should also create a public key class eventually.

Reply:
I was thinking about using the `Signature` class from `crypto` package directly instead. But it's a theme for discussion and for a different PR.

Reply:
Yes, probably a good call. We could add `toBase58()` as an extension only.